src/Symfony/Component/Console/Application.php

```diff
@@ -226,7 +228,18 @@ public function doRun(InputInterface $input, OutputInterface $output)
                 $e = $event->getError();
             }
 
-            throw $e;
+            if (!($e instanceof CommandNotFoundException && !$e instanceof NamespaceNotFoundException) || 1 !== count($alternatives = $e->getAlternatives()) || !$input->isInteractive()) {
+                throw $e;
+            }
+
+            $alternative = $alternatives[0];
+            $question = new ConfirmationQuestion(sprintf("<error>Command \"%s\" is not defined.</error>\n\nDo you want to run \"%s\" instead? [y/n] ", $name, $alternative), false);
+
+            if (!(new QuestionHelper())->ask($input, $output, $question)) {
+                return 1;
```

Comment:
The error event allows to set a custom exit code and override the exception, it must keep working